pkg/plugin/manager.go

```diff
@@ -76,7 +82,16 @@ func (m *Manager) DownloadProviders(ctx context.Context, providers []*config.Req
 			return err
 		}
 		m.providers[providerName] = details
+		traceData[i] = providerName + "@" + details.Version
```

Comment:
would add a method to Plugin Interface to get Provider Full name which should include details.Provider + details.Version, this is also regarding line 76

Reply:
The only problem is `registry.ProviderDetails` doesn't seem to implement the `Plugin` interface.

Reply:
So add a method to `registry.ProviderDetails` ?

Reply:
It has to be the whole interface, so 5/6 methods. It just reads the struct fields without utilizing any interface abstraction. I don't want to touch that in this PR.